sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -116,6 +110,71 @@ object GpuProjectExec extends Arm {
       new ColumnarBatch(newColumns, cb.numRows())
     }
   }
+
+  /**
+   * Just like project, but it will try and retry the operations if it can.
+   * @param sb the input batch
+   * @param boundExprs the expressions to run
+   * @return the resulting batch
+   */
+  def projectAndCloseWithRetrySingleBatch(sb: SpillableColumnarBatch,
```

Comment:
might be nice to clarify this is internal (to be used for a specific tier) as far as I understand.

Reply:
It isn't totally that way.  It can be used stand alone, but the Tiered API I think is better/simpler to use.